### PR TITLE
Return Literal when column is not a Phinx type

### PR DESCRIPTION
**Pull request by martijngastkemper**
This is an alternative for PR #1432 and adds support for all relevant DBMS adapters.

### CODE REVIEW

File: src/Phinx/Db/Adapter/MysqlAdapter.php [hunk 1]
```diff
@@ -1036,8 +1036,12 @@ public function getPhinxType($sqlTypeDef)
                     break;
             }
 
-            // Call this to check if parsed type is supported.
-            $this->getSqlType($type, $limit);
+            try {
+                // Call this to check if parsed type is supported.
+                $this->getSqlType($type, $limit);
+            } catch (\RuntimeException $e) {
```
Comment (martijngastkemper) [inline src/Phinx/Db/Adapter/MysqlAdapter.php]: I don't like this try catch, but `getSqlType` in other adapters throw a `RuntimeException`. Changing this behaviour in the `MysqlAdapter` isn't consistent so I decided to use this way.

Comment (lorenzo) [reply src/Phinx/Db/Adapter/MysqlAdapter.php]: Yeah, it would be nicer if the exception was more specific

**Comment by lorenzo** [comment]
This is a great solution! 

Any change you can add this to the other adapters?

**Comment by martijngastkemper** [comment]
PostgreSQL already had support for `Literal`. I've added it to SQLlite and SQLServer.

**Comment by martijngastkemper** [comment]
codecov/patch and Scrutinizer give weird results. Could it be that some force push commands break them? I had to use it some times to test the SqlServer adapter.

File: src/Phinx/Db/Adapter/MysqlAdapter.php (hunk 1 above)
Comment (martijngastkemper) [reply src/Phinx/Db/Adapter/MysqlAdapter.php]: I'll add a more specific exception.

File: src/Phinx/Db/Adapter/UnsupportedColumnTypeException.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Phinx\Db\Adapter;
+
+/**
+ * Exception thrown when a column type doesn't match a Phinx type.
+ *
+ * @author Martijn Gastkemper <martijn@grrr.nl>
+ */
+class UnsupportedColumnTypeException extends \Exception {
```
Comment (lorenzo) [inline src/Phinx/Db/Adapter/UnsupportedColumnTypeException.php]: Probably extending `RuntimeException` is a good idea to keep other things working as they used to

**Comment by martijngastkemper** [comment]
@lorenzo Took some commits to fix the exception messages, but that's done. Now working on a unrelated but [very strange line in the Postgresql adapter](https://github.com/cakephp/phinx/pull/1436/files#diff-7ab560398b9e11a6639f36d0856ef7f2R1005). The function always returns a string , but in one case it returns an array. The function is used by `getColumns` in the same class which expects a string. But after this change unit tests are failing. So something is going wrong. Maybe you can to elucidate this behaviour. Hopefully I can finish this PR tonight (and do some squashing...).

**Comment by martijngastkemper** [comment]
The result of `getPhinxType` is used [on this line in `PostgressAdapter`](https://github.com/martijngastkemper/phinx/blob/mysql-getcolums-literal-support/src/Phinx/Db/Adapter/PostgresAdapter.php#L416). That will fail because a column type can only by a string or `Literal`.

Right now I'm going to look into this problem.

**Comment by martijngastkemper** [comment]
PR #511 introduced support for PostgreSQL smallint column type. `Literal` is added for these kind of PR's. In this case it's possible to add support for smallint to all adapters. MySQL, SQLite and SqlServer support this column type aswel. Adding support for smallint to all adapters is in my opinion the best solution. Adds a new feature, doesn't break backwards compatibility and it's not very hard to accomplice.

**Comment by martijngastkemper** [comment]
To finish this PR small integer support must be added to all adapters. See PR #1438 .

**Comment by lorenzo** [comment]
just need to resolve the conflicts and we are good to go!

**Comment by martijngastkemper** [comment]
I'll fix it tonight!

**Comment by martijngastkemper** [comment]
@lorenzo I've resolved the conflicts and squashed the ugly commits. The codecov check didn't pass but when looking at the details nothings shows up. Can you fix that or tell me what I have to do to let the check pass?

**Comment by lorenzo** [comment]
@martijngastkemper Thanks for all your work! I think it is safe to ignore codecov in this case

**Pull request merged**